### ISSUE TITLE
Duplicating type bounds results in them being ignored entirely

### ISSUE THREAD

**Issue by RundownRhino** [intellij-rust/intellij-rust#8416]
## Environment

* **IntelliJ Rust plugin version: 0.4.162.4321-212**
* **Rust toolchain version: 1.56.0**
* **IDE name and version: IntelliJ IDEA 2021.2.3 (Community Edition)**
* **Operating system: Windows 7**

## Problem description
When some bound, such as Copy, is specified in a generic twice (once directly and once implied by another bound), this seems to result in this bound not being considered at all. See example below.

## Steps to reproduce

```rust
struct A<T>(T) where T: Copy;

trait Id: Copy {
    fn id(self) -> Self;
}

trait Problem {
    fn problem(self) -> (Self, Self) where Self: Sized;
}

impl<T> Problem for A<T> 
where T: Id + Copy // Note how Id already implies Copy
{
    fn problem(self) -> (Self, Self) {
        let a = self.0.id();
        (Self(a), Self(a)) // <== use of moved value on second usage of a, even though it's a T: Copy
    }
}
```
This compiles without warnings. The Rust plugin, meanwhile, shows a "use of moved value" error on second usage of `a` in `problem`. The issue vanishes if `+ Copy` is removed from the bound of `T` in the impl.

**Comment by afetisov** [comment]
This seems to be fixed. Close?

Also, a duplicate of #8439

**Comment by ortem** [comment]
@vlad20012 WDYT?

**Comment by vlad20012** [comment]
Yes, I fixed this in #8902
